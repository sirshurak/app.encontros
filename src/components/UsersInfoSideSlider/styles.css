#users-info-side-slider {
    min-height: 180px;
}