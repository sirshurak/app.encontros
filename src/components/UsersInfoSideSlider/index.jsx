import React from "react";
import SideSlider from "../SideSlider";
import UserInfo from "../UserInfo";
import UserInfoClick from "../UserInfoClick";

import "./styles.css";

function UsersChatSideSlider({ users, title, ageTitle }) {
  return (
    <div id="users-info-side-slider">
      <SideSlider title={title}>
        {users?.length &&
          users.map((user, key) => (
            <UserInfoClick user={user} key={key}>
              <UserInfo {...user} variant="horizontal" ageTitle={ageTitle} />
            </UserInfoClick>
          ))}
      </SideSlider>
    </div>
  );
}

export default UsersChatSideSlider;
