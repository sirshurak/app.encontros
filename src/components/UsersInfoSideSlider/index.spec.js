import React from 'react';
import { shallow, mount } from 'enzyme';
import UsersInfoSideSlider from './index';
import AppContext from '../../contexts/AppContext';

const users = [{
    photo: 'https://picsum.photos/200/?image=26',
    username: 'username',
    city: 'Test City',
    gender: 'Man',
    sexualOrientation: 'Heterosexual',
    age: 27
}];

describe('UsersInfoSideSlider component', () => {
    it('should match shallow snapshop', () => {
        const component = shallow(<UsersInfoSideSlider title="Title" ageTitle="anos" users={users}/>)
        expect(component).toMatchSnapshot();
    });

    it('should mount slider items', () => {
        const changePage = jest.fn();
        const component = mount(
            <AppContext.Provider value={{ changePage }}>
                <UsersInfoSideSlider title="Title" ageTitle="anos" users={users}/>
            </AppContext.Provider>
        );

        expect(component.find('h3').text()).toBe('username');
    });
});