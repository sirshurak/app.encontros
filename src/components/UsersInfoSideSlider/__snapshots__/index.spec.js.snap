// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersInfoSideSlider component should match shallow snapshop 1`] = `
<div
  id="users-info-side-slider"
>
  <SideSlider
    title="Title"
  >
    <UserInfoClick
      key="0"
      user={
        Object {
          "age": 27,
          "city": "Test City",
          "gender": "Man",
          "photo": "https://picsum.photos/200/?image=26",
          "sexualOrientation": "Heterosexual",
          "username": "username",
        }
      }
    >
      <UserInfo
        age={27}
        ageTitle="anos"
        city="Test City"
        gender="Man"
        photo="https://picsum.photos/200/?image=26"
        sexualOrientation="Heterosexual"
        username="username"
        variant="horizontal"
      />
    </UserInfoClick>
  </SideSlider>
</div>
`;
